### PR TITLE
Laurent-Olivier Labeis Cashflow generation for settlement payout

### PR DESCRIPTION
**Pull request by regnosys-prod-user**

### CODE REVIEW

**Review by manucarreramoreno — APPROVED** [review]

File: rosetta-source/src/main/rosetta/event-common-type.rosetta [hunk 1]
```diff
@@ -211,7 +211,10 @@ type TransferState: <"Defines the fundamental financial information associated w
     transfer Transfer (1..1) <"Represents the Transfer that has been effected by a business or life-cycle event.">
     transferStatus TransferStatusEnum (0..1) <"Represents the State of the Transfer through its life-cycle.">
 
-type Transfer extends TransferBase: <"Defines the movement of an Asset (eg cash, securities or commodities) between two parties on a date.">
+type Transfer extends CashflowBase: <"Defines the movement of an Asset (eg cash, securities or commodities) between two parties on a date.">
```
[the remark this answers is not in the document]
Comment (manucarreramoreno) [reply rosetta-source/src/main/rosetta/event-common-type.rosetta:214]: I think CashflowBase now allows to represent securities. Also TransferBase has been deleted

[the remark this answers is not in the document]
Comment (chrisisla) [reply rosetta-source/src/main/rosetta/event-common-type.rosetta:214]: If a transfer is performed on a non-cash trade (i.e. using other securities as collateral and not cash) then there is no cash so this would not affect the cash flow. 

I'm more querying the change from TransferBase to CashflowBase as Transfer is not only used for cash transfers?

[the remark this answers is not in the document]
Comment (lolabeis) [reply rosetta-source/src/main/rosetta/event-common-type.rosetta:214]: @chrisisla Thank you for your review.

You are right, `CashflowBase` does represent more than the transfer of cash, it represents the transfer of any asset. And so does `Cashflow` now.

Aligning the name of the common base class to `Cashflow` rather than `Transfer` isn't perfect and results from a trade-off. This base class is now in the (lower level) `product` rather than the (higher level) `event` namespace, but the former doesn't know about the `Transfer` concept. In other words, it would have been strange to position `TransferBase` in `product` and `Transfer` in `event`.

What about potential alternatives, and their cons?
* Migrate `Transfer` and all its associated concepts in `product` --> the scope creep is worrying, although maybe that's the right target state?
* Rename `Cashflow` to something more generic, now that we're extending its behaviour (`AssetFlow`?) --> this would be an "invented" name, and possibly duplicative vs. "transfer". It was raised at the DPBE, but inconclusive.

In short, we remain a bit saddled by the legacy of `Transfer` and `Cashflow` having been developed separately, but hope this goes in the right direction. Further inputs welcome :-)

[the remark this answers is not in the document]
Comment (chrisisla) [reply rosetta-source/src/main/rosetta/event-common-type.rosetta:214]: HI @lolabeis ,
Apologies, I missed your Issue explaining this change, and I can't attend the DPBE and could not attend the CRWG earlier this week as I was involved in a different cross association call. So this is the first time I have seen this proposed change.
By extending CashFlowBase we are now saying that Transfer is a type of Cashflow[Base], which it won't be if there is no cash, which is the most common type of stock loan. I've read through your powerpoint and I'm also still not really clear as to why we need this change at all? A Transfer is an event so why does it need to move to product?

[the remark this answers is not in the document]
Comment (lolabeis) [reply rosetta-source/src/main/rosetta/event-common-type.rosetta]: Hi @chrisisla Following our conversation last week:
I have adjusted this contribution and replaced `CashflowBase` by `AssetFlowBase` - although I have kept the name `Cashflow` unchanged (to minimise impact). This means `Transfer` now extends `AssetFlowBase`.
Can you please confirm that you're happy with this adjustment?

[the remark this answers is not in the document]
Comment (chrisisla) [reply rosetta-source/src/main/rosetta/event-common-type.rosetta]: I'm not a massive fan of `AssetFlowBase` either but it is more aligned to what we're describing than `CashFlowBase` was!

**Review by chrisisla — APPROVED** [review]

File: rosetta-source/src/main/rosetta/event-common-type.rosetta (hunk 1 above)
[the remark this answers is not in the document]
Comment (lolabeis) [reply rosetta-source/src/main/rosetta/event-common-type.rosetta]: Thanks @chrisisla!
